client/src/analytics_component.rs

```diff
@@ -1,68 +1,520 @@
+                    }
+                } else {
+                    ""
+                };
+
+                let onclick = {
+                    let sort_field = sort_field.clone();
+                    let sort_direction = sort_direction.clone();
+                    let field = field.clone();
+                    Callback::from(move |_| {
+                        if &field == &*sort_field {
+                            sort_direction.set(match *sort_direction {
+                                SortDirection::Ascending => SortDirection::Descending,
+                                SortDirection::Descending => SortDirection::Ascending,
+                            });
+                        } else {
+                            sort_field.set(field.clone());
+                            sort_direction.set(SortDirection::Ascending);
+                        }
+                    })
+                };
+
+                html! {
+                    <th class={if is_active { "active" } else { "" }} {onclick}>
+                        <span class="sort-header">
+                            { label }
+                            <span class="sort-indicator">{ direction }</span>
+                        </span>
+                    </th>
+                }
+            };
+
             Ok(html! {
                 <div class="analytics">
-                    <table>
-                        <thead>
-                            <tr>
-                                <th>{"IP Address"}</th>
-                                <th>{"URL Path"}</th>
-                                <th>{"Country"}</th>
-                                <th>{"Count"}</th>
-                            </tr>
-                        </thead>
-                        { analytics }
-                    </table>
+                    <AnalyticsHeader />
+                    <AnalyticsToolbar
+                        on_filter_change={Callback::from(move |c| country_filter.set(c))}
+                        on_search_change={
+                            let search_filter = search_filter.clone();
+                            Callback::from(move |s| search_filter.set(s))
+                        }
+                        on_view_change={
+                            let view_type = view_type.clone();
+                            Callback::from(move |v| view_type.set(v))
+                        }
+                        on_export={
+                            let filtered_data = filtered_data.clone();
+                            Callback::from(move |_| {
+                                let csv = export_to_csv(&filtered_data);
+                                download_csv(&csv);
+                            })
+                        }
+                        view_type={(*view_type).clone()}
+                        data={filtered_data.clone()}
+                    />
+                    {
+                        if *view_type == ViewType::Table {
```

Comment:
My brain is melting looking at this macro. I think this ould be great to pull out into it's own compenent.